pkg/util/cluster/cluster.go

```diff
@@ -0,0 +1,48 @@
+package cluster
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+const OpenShift3Major = 3
+
+type OpenShiftVersion struct {
+	Major string `json:"major"`
+	Minor string `json:"minor"`
+}
+
+func IsOnOpenShift3() (bool, error) {
```

Comment:
The `virt-operator` already contains logic which detects reliably if we are on openshift. As far as I know `/version/openshift` does not exist anymore in OCP4.

Reply:
I need to know the version, OpenShift3 has a different cpu-manager path than OpenShift4 (the same as k8s), the check from virt-operator is not helpful here (it checks only if security context constraints exists which is available on OpenShift4 as well)

Reply:
There is also code in virt-operator which helps you finding that out. Also virt-operator has to distinguish between ok3 and okd4 in some places. Afaik it does something like, "if on openshift and k8s version is bigger than X, then it is okd4`".

Reply:
..and you're right about OCP4 `version` endpoint, it doesn't exist, as far as i remember there was an issue to bring it back, not sure about the decision though.

Anyway, I need to know that I am on OpenShift3, k8s and ocp4 have the same file path.

Reply:
Ok, I will check it.

Reply:
Well, it's not exactly in `virt-operator` pkg but in tests, but I will get the solution from it and refactor a bit. https://github.com/kubevirt/kubevirt/blob/master/tests/utils.go#L3436